File: Matlab/+ACME/Color/color2normal.m
function [N] = color2normal(C)
% COLOR2NORMAL  Converts a given color vector into normal directions.
%   N = COLOR2NORMAL(C) returns a nx3 vector.
if( ~isdouble(C) )
    C = color2double(C);
end
N = (C .* 2)-1;
end

File: Python/ACME/encoding/color2normal.py
import torch
from color.color2float import *

def color2normal(C):
    """
    Converts a given color into a normal direction

    Parameters
    ----------
    C : Tensor
        a color (n,3) tensor

    Returns
    -------
    Tensor
        a normal tensor
    """

    return torch.add(torch.mul(color2float(C),2),-1)
